server/src/main/scala/services/graphstages/Crunch.scala

```diff
@@ -219,53 +223,90 @@ object Crunch {
     desks(date.getHourOfDay)
   }
 
+  def purgeExpired[A <: WithTimeAccessor, B](expireable: SortedMap[A, B], now: () => SDateLike, expireAfter: Int): SortedMap[A, B] = {
+    val start = SDate.now().millisSinceEpoch
+    val thresholdMillis = now().addMillis(-1 * expireAfter).millisSinceEpoch
+    val fresh = expireable.dropWhile {
+      case (a: A, _) => a.timeValue < thresholdMillis
+    }
+    println(s"Efficient purge took ${SDate.now().millisSinceEpoch - start}ms (${expireable.size} element SortedMap)")
```

Comment:
do we want to log this properly or delete it?

Reply:
Deleted